pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -890,6 +844,14 @@ func (a *Actuator) requestPowerOn(ctx context.Context, baremetalhost *bmh.BareMe
 // deleteMachineNode deletes the node that mapped to specified machine
 func (a *Actuator) deleteNode(ctx context.Context, node *corev1.Node) error {
 	if !node.DeletionTimestamp.IsZero() {
+		// remove finalizer
+		if utils.StringInList(node.Finalizers, nodeFinalizer) {
+			node.Finalizers = utils.FilterStringFromList(node.Finalizers, nodeFinalizer)
+			if err := a.client.Update(ctx, node); err != nil {
+				log.Printf("Failed to remove node finalizer from %s, error: %s", node.Name, err.Error())
+				return err
+			}
```

Comment:
This seems like duplication of the code in `removeNodeFinalizer()`.
we can separate `removeNodeFinalizer()` into two functions (one with node argument and one with machine).

also - if we call `removeNodeFinalizer()` from Update/Create, this is probably redundant and can be removed.